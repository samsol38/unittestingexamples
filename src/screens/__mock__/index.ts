const mockItems = [
  {id: 1, name: 'Item 1'},
  {id: 2, name: 'Item 2'},
  {id: 3, name: 'Item 3'},
];

const mockUserData = {
  results: [
    {
      name: {
        first: 'John',
        last: 'Doe',
      },
      email: 'john.doe@example.com',
    },
  ],
};

export {mockItems, mockUserData};
