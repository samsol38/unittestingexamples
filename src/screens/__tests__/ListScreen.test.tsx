// ListScreen.test.tsx
import React from 'react';
import {render} from '@testing-library/react-native';
import ListScreen from '../list';

import {mockItems} from '../__mock__';

describe('ListScreen render with snapshot', () => {
  test('matches snapshot', () => {
    const tree = render(<ListScreen items={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ListScreen render with snapsho with black itemst', () => {
  test('matches snapshot', () => {
    const tree = render(<ListScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ListScreen', () => {
  test('renders screen component without items', () => {
    const {getByText} = render(<ListScreen />);
    const componentText = getByText('ListScreen Component');
    expect(componentText).toBeDefined();
  });

  test('renders screen component with black items', () => {
    const {getByText} = render(<ListScreen items={[]} />);
    const componentText = getByText('ListScreen Component');
    expect(componentText).toBeDefined();
  });

  test('renders item names correctly', () => {
    const {getByText} = render(<ListScreen items={mockItems} />);
    mockItems.forEach(item => {
      const itemText = getByText(item.name);
      expect(itemText).toBeDefined();
    });
  });

  test('renders correct number of items', () => {
    const {getAllByText} = render(<ListScreen items={mockItems} />);
    const itemTexts = getAllByText(/Item/);
    expect(itemTexts.length).toBe(mockItems.length);
  });
});
