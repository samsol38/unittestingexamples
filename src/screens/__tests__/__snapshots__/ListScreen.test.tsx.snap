// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListScreen render with snapsho with black itemst matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingVertical": 50,
    }
  }
>
  <Text>
    ListScreen Component
  </Text>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</View>
`;

exports[`ListScreen render with snapshot matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingVertical": 50,
    }
  }
>
  <Text>
    ListScreen Component
  </Text>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</View>
`;
